### PR TITLE
Several small fixes

### PR DESCRIPTION
**Pull request by leeping**
In `forcefield.py`:
- Fix a .ffxml parsing bug in which leading whitespace causes a floating point number to not be recognized.
- The OpenFF unit string is incompatible with evaluated parameters (evaluated parameters are not independently optimized, but are functions of other parameters).  As a temporary fix, units have been disabled for evaluated parameters. @j-wags : Your input could be helpful here, would OpenFF ever want evaluated parameters with units?
- Updated the OpenMM interface to incorporate the new method for obtaining names of Force objects while preserving compatibility with OpenMM <7.6.
- Fixed `CopyAmoebaVdwParameters` such that type parameters and type pair parameters are copied.
- Added the ability to replace parameter tag names with more descriptive ones when building parameter names (such as `<Angle .../>` in `AmoebaOutOfPlaneBendForce`

In `objective.py`:
- Fixed remove micro-iteration folder feature. The previous code never worked.

In `studies`:
- Changed iAMOEBA hydrogen vdW parameter sigma to 1.0 (for compatibility with OpenMM >= 7.6).

TODO:
- Undo unintended change to `make-vdata.py`.

### CODE REVIEW

**Comment by pavankum** [comment]
Ahh sorry, I see that you already made required changes, please ignore the previous comment.

**Review by j-wags — APPROVED** [review]
This should be good to merge if you're OK with the amoeba test beginning to fail (not sure about all the context there so it's your call)

* (not blocking, commented in this review) One stray debug statement left over.
* (your call) The CI is showing a [new error](https://github.com/leeping/forcebalance/actions/runs/4017686403/jobs/6902352143#step:9:950): `FAILED src/tests/test_engine.py::TestAmoebaWater6::test_interaction_energy - IndexError: list index out of range`

> The OpenFF unit string is incompatible with evaluated parameters (evaluated parameters are not independently optimized, but are functions of other parameters). As a temporary fix, units have been disabled for evaluated parameters. @j-wags : Your input could be helpful here, would OpenFF ever want evaluated parameters with units?

"Ever" is a big word, but if it hasn't been working since unit-bearing values in SMIRNOFF started being written as full quantities (around 2019?) then I think we're good without it.

File: src/openmmio.py
Comment (leeping) [reply src/openmmio.py:1183]: > This should be good to merge if you're OK with the amoeba test beginning to fail (not sure about all the context there so it's your call)

I hadn't checked that the test was failing.  Thanks a lot, and I will look into it.

> "Ever" is a big word, but if it hasn't been working since unit-bearing values in SMIRNOFF started being written as full quantities (around 2019?) then I think we're good without it.

I should clarify, I'm not asking you to commit to never having this feature!  What I meant was, if we ever want to have this feature, we may need to think about the design a bit.  Perhaps FB could split off the unit string of the "initial" evaluated parameter and append it to the computed value, or something like that.

Comment (j-wags) [reply src/openmmio.py:1183]: Ah, yeah, I'd seen some other code that assumes the new magnitude has the same units as the original value in my PR from earlier. I think it's a bit dangerous (could invite edge cases, eg regularization getting confused if similarly-named fields have different units), so I wouldn't put that logic in more places unless there's a driving use case for it. 

A bigger restructuring might try carrying quantities around internally everywhere, but that's well beyond the scope of a 1.9.5 release. We may experiment with that in the OpenFF fork of forcebalance and could report back if we get it working.

**Pull request merged**
